sktime/forecasting/auto_reg.py

```diff
@@ -0,0 +1,273 @@
+        using observations 3,...,nobs. hold_back must be >= the maximum lag in
+        the model.
+    period : {None, int}
+        The period of the data. Only used if seasonal is True. This parameter
+        can be omitted if using a pandas object for endog that contains a
+        recognized frequency.
+    missing : str
+        Available options are 'none', 'drop', and 'raise'. If 'none', no nan
+        checking is done. If 'drop', any observations with nans are dropped.
+        If 'raise', an error is raised. Default is 'none'.
+    deterministic : DeterministicProcess
+        A deterministic process.  If provided, trend and seasonal are ignored.
+        A warning is raised if trend is not "n" and seasonal is not False.
+    old_names : bool
+        Flag indicating whether to use the v0.11 names or the v0.12+ names.
+
+        .. deprecated:: 0.13.0
+
+           old_names is deprecated and will be removed after 0.14 is
+           released. You must update any code reliant on the old variable
+           names to use the new names.
+
+    Examples
+    --------
+    Use AutoREG to forecast univariate data.
+    >>> from sktime.forecasting.auto_reg import AutoREG  # doctest: +SKIP
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> data = load_airline()
+    >>> autoreg_sktime = AutoREG(lags=2, trend="c")  # doctest: +SKIP
+    >>> autoreg_sktime.fit(y=data)  # doctest: +SKIP
+    AutoREG(lags=2)
+    >>> fh = ForecastingHorizon([x for x in range(1, 13)])
+    >>> y_pred = autoreg_sktime.predict(fh=fh)  # doctest: +SKIP
+
+
+    Use AutoREG to forecast with exogenous data.
+    >>> from sktime.forecasting.auto_reg import AutoREG  # doctest: +SKIP
+    >>> from sktime.datasets import load_longley
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> y, X_og = load_longley()
+    >>> X_oos = X_og.iloc[-5:, :]
+    >>> y, X = y.iloc[:-5], X_og.iloc[:-5, :]
+    >>> X, X_oos = X[["GNPDEFL", "GNP"]], X_oos[["GNPDEFL", "GNP"]]
+    >>> autoreg_sktime = AutoREG(lags=2, trend="c")  # doctest: +SKIP
+    >>> autoreg_sktime.fit(y=y, X=X)  # doctest: +SKIP
+    AutoREG(lags=2)
+    >>> fh = ForecastingHorizon([x for x in range(1, 4)])
+    >>> y_pred = autoreg_sktime.predict(X=X_oos, fh=fh)  # doctest: +SKIP
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "scitype:y": "univariate",
+        "ignores-exogeneous-X": False,
+        "requires-fh-in-fit": False,
+        "python_version": None,
+        "python_dependencies": "statsmodels>=0.13.0",
+    }
+
+    # todo: add any hyper-parameters and components to constructor
+    def __init__(
+        self,
+        lags=None,
+        trend="c",
+        seasonal=False,
+        hold_back=None,
+        period=None,
+        missing="none",
+        deterministic=None,
+        old_names=False,
```

Comment:
As per the documentation, this is deprecated. I think it'd be better to avoid adding this.

Ref. https://www.statsmodels.org/stable/generated/statsmodels.tsa.ar_model.AutoReg.html

Reply:
Thanks, for some reason we initially read that as "will be depreciated" not "is already depreciated". Fixed in upcoming push.